permissions/src/main/java/org/openmbee/sdvc/permissions/config/PermissionInit.java

```diff
@@ -54,15 +54,18 @@ public void onApplicationEvent(final ApplicationReadyEvent event) {
             }
         }
 
-        for (Role role : roleRepo.findAll()) {
+        List<Role> roleList = roleRepo.findAll();
+        List<Privilege> privList = privRepo.findAll();
+
+        for (Role role : roleList) {
             Set<Privilege> pSet = new HashSet<Privilege>();
-            for (Privilege priv : privRepo.findAll()) {
+            for (Privilege priv : privList) {
                 if (((RPmap.get(role.getName())).contains(priv.getName())) && ((role.getPrivileges() == null) || (!(role.getPrivileges().contains(priv))))) {
                     pSet.add(priv);
-                    role.setPrivileges(pSet);
-                    roleRepo.saveAndFlush(role);
                 }
             }
+            role.setPrivileges(pSet);
```

Comment:
this only sets priv that are not already there and removes ones that are already there - you can either remove the entire second conditional clause, or add the priv to the existing role.getPrivileges instead of a brand new set